### PR TITLE
Respect the `decodeEntities` flag with rgxp => url

### LINKED ISSUE #3401: rgxp->url encodes even when decodeEntities is set to true
**Affected version(s)**

All

**Description**

If I configure a field with `eval->rgxp = 'url'` as well as `eval->decodeEntities = 'true'` I end up having encoded `&amp;`'s in the database.

This should probably be checked here https://github.com/contao/contao/blob/bdc6e30cf50421bf3ec98d67b0b21ddd5ffa950d/core-bundle/src/Resources/contao/library/Contao/Widget.php#L989

### PR DESCRIPTION
**Pull request by rabauss**
| Q                | A
| -----------------| ---
| Fixed issues     | Fixes #3401

I just added the suggestion from @ausi here https://github.com/contao/contao/issues/3401#issuecomment-908367764
Although I'm not really sure about the correct position, but a first test fixed the problem!

### CODE REVIEW

**Review by ausi — DISMISSED** [review]

File: core-bundle/src/Resources/contao/library/Contao/Widget.php [hunk 1]
```diff
@@ -982,6 +982,10 @@ protected function validator($varInput)
 
 				case 'url':
 					$varInput = StringUtil::specialcharsUrl($varInput);
+					
+					if ($this->decodeEntities) {
```
Comment (ausi) [inline core-bundle/src/Resources/contao/library/Contao/Widget.php]:
```suggestion
					if ($this->decodeEntities)
					{
```
Legacy coding style ☺️

**Review by ausi — DISMISSED** [review]

**Review by Toflar — DISMISSED** [review]

**Comment by leofeyer** [comment]
@rabauss Can you please fix the CI chain? I cannot merge the PR otherwise.

**Comment by rabauss** [comment]
Ohm, I rebased but now I get CS errors - although there is nothing to fix in my local environment!

**Review by ausi — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @rabauss.

**Pull request merged**